src/components/settings/PushNotifications/GlobalPushNotifications.tsx

```diff
@@ -0,0 +1,346 @@
+import {
+  Grid,
+  Paper,
+  Typography,
+  Checkbox,
+  Button,
+  Divider,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+} from '@mui/material'
+import { Fragment, useEffect, useMemo, useState } from 'react'
+import type { ReactElement } from 'react'
+
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { sameAddress } from '@/utils/addresses'
+import useChains from '@/hooks/useChains'
+import { useAppSelector } from '@/store'
+import { useNotificationPreferences } from './hooks/useNotificationPreferences'
+import { useNotificationRegistrations } from './hooks/useNotificationRegistrations'
+import { selectAllAddedSafes } from '@/store/addedSafesSlice'
+import { trackEvent } from '@/services/analytics'
+import { PUSH_NOTIFICATION_EVENTS } from '@/services/analytics/events/push-notifications'
+import { requestNotificationPermission } from './logic'
+import type { NotifiableSafes } from './logic'
+import type { AddedSafesState } from '@/store/addedSafesSlice'
+import type { NotificationPreferences } from '@/services/firebase/preferences'
+import CheckWallet from '@/components/common/CheckWallet'
+
+import css from './styles.module.css'
+
+// Convert data structure of added Safes
+export const transformAddedSafes = (addedSafes: AddedSafesState): NotifiableSafes => {
+  const obj: NotifiableSafes = {}
+
+  for (const [chainId, addedSafesOnChain] of Object.entries(addedSafes)) {
+    obj[chainId] = Object.keys(addedSafesOnChain)
+  }
+
+  return obj
+}
+
+// Convert data structure of currently notified Safes
+const transformCurrentSubscribedSafes = (allPreferences: NotificationPreferences): NotifiableSafes => {
+  const obj: NotifiableSafes = {}
+
+  for (const { chainId, safeAddress } of Object.values(allPreferences)) {
+    if (!obj[chainId]) {
+      obj[chainId] = []
+    }
+
+    obj[chainId].push(safeAddress)
+  }
+
+  return obj
```

Comment:
In both of these a `reduce` would be a little easier to follow.

Reply:
Adjusted accordingly in 75b5a63d8189dbe37b42ffb469b1bb2361a2d28a.